algo/src/main/java/org/neo4j/graphalgo/impl/louvain/Louvain.java

```diff
@@ -209,15 +209,18 @@ public long getCommunityCount() {
                 .mapToObj(i -> new Result(i, communities[i]));
     }
 
-    public Stream<StreamingResult> dendrogramStream() {
+    public Stream<StreamingResult> dendrogramStream(boolean includeIntermediateCommunities) {
         return IntStream.range(0, rootNodeCount)
                 .mapToObj(i -> {
-                    List<Long> communities = new ArrayList<>(dendrogram.length);
-                    for (int[] community : dendrogram) {
-                        communities.add((long) community[i]);
+                    List<Long> communitiesList = null;
+                    if (includeIntermediateCommunities) {
```

Comment:
Maybe just a cosmetic fix, but won't the community ids of the last two iterations always be the same as the algo checks if it has converged and if the community ids don't change in the last iteration it stops. So we can skip returning the last value of array as it will be always the same as second last value.

Reply:
@mknblch will know for sure so Martin what do you say?

Reply:
tested it on another dataset and it looks like it is not always the same, so please just disregard this